godot-core/src/engine.rs

```diff
@@ -16,6 +16,40 @@ pub use crate::gen::central::global;
 pub use crate::gen::classes::*;
 pub use crate::gen::utilities;
 
+use self::packed_scene::GenEditState;
+
+/// Extension trait for convenience functions on `PackedScene`
+pub trait PackedSceneExt {
+    /// Instantiates the scene as type `T`, panicking if not found or bad type.
+    ///
+    /// # Panics
+    /// If the scene is not type `T` or inherited.
+    fn instantiate_as<T>(&self) -> Gd<T>
+    where
+        T: GodotClass + Inherits<Node>,
+    {
+        self.try_instantiate_as::<T>().unwrap()
```

Comment:
Instead of `unwrap()`, could you add a descriptive error message taking into account the variables?

For prior art, see
https://github.com/godot-rust/gdext/blob/7573721b8a8c633600494e9137f73792722bf89f/godot-core/src/obj/gd.rs#L316-L331